library/psa_crypto_cipher.c

```diff
@@ -563,15 +564,19 @@ psa_status_t mbedtls_psa_cipher_decrypt(
             goto exit;
     }
 
-    status = mbedtls_psa_cipher_update( &operation, input + operation.iv_length,
+    status = mbedtls_psa_cipher_update( &operation,
+                                        ( input == NULL ? NULL :
+                                          input + operation.iv_length ),
                                         input_length - operation.iv_length,
                                         output, output_size, &olength );
     if( status != PSA_SUCCESS )
         goto exit;
 
     accumulated_length = olength;
 
-    status = mbedtls_psa_cipher_finish( &operation, output + accumulated_length,
+    status = mbedtls_psa_cipher_finish( &operation,
+                                        ( output == NULL ? NULL :
+                                          output + accumulated_length ),
```

Comment:
As with the introduction of `mpi_sint_abs()`, the obvious question here is: since this construct is repeated four times, is it worth abstracting it, either with a function or a macro?

Reply:
I'm on the fence. This is considerably simpler than `mpi_sint_abs`. It feels like giving it a meaningful name will make the code longer and won't really be more readable.

Are you happy with seeing the code, or are you going “WTF why do we need a special case for NULL”? If the latter, then a macro/function with documentation that explains why it's needed sometimes is definitely warranted.

Reply:
IMO it's clear that it would be meaningless to compute an address based on NULL + offset, and so having a special case makes sense, but less obvious that it's UB (as opposed to simply a meaningless result) without the special case. So I think either way would be fine.

Reply:
A helper function would be useful here indeed

Reply:
This part is from https://github.com/Mbed-TLS/mbedtls/pull/6609 and I won't change it again here. The pull request here is to arrange for such bugs to be caught by our CI.